### ISSUE TITLE
UnicodeError: label empty or too long

### ISSUE THREAD

**Issue by basst85** [m0rtem/CloudFail#5]
I'm getting this error:

```
/lib/python3.6/encodings/idna.py", line 165, in encode
    raise UnicodeError("label empty or too long")
UnicodeError: label empty or too long

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "cloudfail.py", line 266, in <module>
    subdomain_scan(args.target)
  File "cloudfail.py", line 167, in subdomain_scan
    target_http = requests.get("http://"+subdomain)
  File "/tmp/env/lib/python3.6/site-packages/requests/api.py", line 72, in get
    return request('get', url, params=params, **kwargs)
  File "/tmp/env/lib/python3.6/site-packages/requests/api.py", line 58, in request
    return session.request(method=method, url=url, **kwargs)
  File "/tmp/env/lib/python3.6/site-packages/requests/sessions.py", line 502, in request
    resp = self.send(prep, **send_kwargs)
  File "/tmp/env/lib/python3.6/site-packages/requests/sessions.py", line 612, in send
    r = adapter.send(request, **kwargs)
  File "/tmp/env/lib/python3.6/site-packages/requests/adapters.py", line 440, in send
    timeout=timeout
  File "/tmp/env/lib/python3.6/site-packages/urllib3/connectionpool.py", line 600, in urlopen
    chunked=chunked)
  File "/tmp/env/lib/python3.6/site-packages/urllib3/connectionpool.py", line 356, in _make_request
    conn.request(method, url, **httplib_request_kw)
  File "/usr/lib/python3.6/http/client.py", line 1239, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/usr/lib/python3.6/http/client.py", line 1285, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "/usr/lib/python3.6/http/client.py", line 1234, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/usr/lib/python3.6/http/client.py", line 1026, in _send_output
    self.send(msg)
  File "/usr/lib/python3.6/http/client.py", line 964, in send
    self.connect()
  File "/tmp/env/lib/python3.6/site-packages/urllib3/connection.py", line 166, in connect
    conn = self._new_conn()
  File "/tmp/env/lib/python3.6/site-packages/urllib3/connection.py", line 141, in _new_conn
    (self.host, self.port), self.timeout, **extra_kw)
  File "/tmp/env/lib/python3.6/site-packages/urllib3/util/connection.py", line 60, in create_connection
    for res in socket.getaddrinfo(host, port, family, socket.SOCK_STREAM):
  File "/usr/lib/python3.6/socket.py", line 743, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
UnicodeError: encoding with 'idna' codec failed (UnicodeError: label empty or too long)```

**Comment by m0rtem** [comment]
Hi thanks for reporting this issue, can I just get a bit more info regarding this issue?

What domain are you trying to scan? Have you modified the subdomain list?

**Comment by basst85** [comment]
I'm running this command:
``` python cloudfail.py --target geenstijl.nl --tor```

I've not modified the subdomains list.

**Comment by shello** [comment]
I also saw this error with the default data files from revision 125169c.

It turns out it fails when trying to request `m..<target>` (an invalid address, notice the two periods with an empty string between them). The `m.` entry is in the `data/subdomains.txt` file (line 1275), and as of that commit it's the only entry ending in a period.

**Comment by m0rtem** [comment]
Brilliant find @shello that is most likely the cause of this error (although I am not too sure why it seems to work sometimes) but I will most likely remove this subdomain in the next commit.

**Comment by anhkhoa14592** [comment]
So... the solution is removing "m." at line 1275. I worked for me. Thanks

**Comment by m0rtem** [comment]
This should now be resolved as of commit a9ef9dc95e972ed7992b7db54fd1abd1406f6349
